homeassistant/components/fritz/switch.py

```diff
@@ -287,7 +289,17 @@ def _is_tracked(mac: str) -> bool:
         data_fritz.profile_switches[router.unique_id] = set()
 
     for mac, device in router.devices.items():
-        if device.ip_address == "" or _is_tracked(mac):
+        reason: str | None = None
+        if device.ip_address == "":
+            reason = "Missing IP"
+        elif _is_tracked(mac):
+            reason = "Already tracked"
+        elif pref_disable_new_entities:
+            reason = "Disabled System Options"
+        _LOGGER.debug(
+            "Skip adding device %s [%s], reason: %s", device.hostname, mac, reason
+        )
+        if reason:
```

Comment:
this seems to be duplicate to the above. Maybe we could write a common function for it?

Reply:
Refactored, as well some previously duplicate code from you ;-)